addons/mail/static/src/components/activity/activity.xml

```diff
@@ -130,6 +130,7 @@
                                     <i class="fa fa-upload"/> Upload Document
                                 </button>
                                 <FileUploader
+                                    t-if="activity.thread"
```

Comment:
@robodoo r-

This condition does not hurt in itself, but it is hiding the real issue, which is that an activity should never exist without its thread.

The `replace` in `refreshActivities` is assuming that activities will be deleted, but the field is missing isCausal on thread so it never happens.